GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -229,6 +229,18 @@ public virtual void Shutdown()
             this.indexParsingThread.Wait();
         }
 
+        public void ReleaseIndexParsingRequestWait()

```

Comment:
This name is still very confusing. My problem with it is that it assumes the caller knows or cares about the internal details of how this is going to work - but the caller should just care about what they want done, not how. I would suggest something like `ParseIndex` or `UpdateProjection` or something to indicate the action it will be taking, rather than the mechanism.